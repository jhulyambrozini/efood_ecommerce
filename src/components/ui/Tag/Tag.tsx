import { TagContainer } from './styles'

type TagProps = {
  children: string
}

const Tag = ({ children }: TagProps) => <TagContainer>{children}</TagContainer>

export default Tag
