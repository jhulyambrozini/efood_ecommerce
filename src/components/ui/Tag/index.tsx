import { TagContainer } from './styles'

export type TagProps = {
  text: string
}

const Tag = ({ text }: TagProps) => <TagContainer>{text}</TagContainer>

export default Tag
