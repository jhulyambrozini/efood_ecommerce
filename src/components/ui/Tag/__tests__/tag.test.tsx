import '@testing-library/jest-dom'
import { screen } from '@testing-library/dom'
import { render } from '@testing-library/react'

import Tag from '..'

describe('<Tag />', () => {
  it('should be render with the text "destaque"', () => {
    render(<Tag text="destaque" />)

    expect(screen.findByRole('tag-container', { name: /destaque/i }))
  })
  it('should match snapshot', () => {
    const { container } = render(<Tag text="destaque" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
