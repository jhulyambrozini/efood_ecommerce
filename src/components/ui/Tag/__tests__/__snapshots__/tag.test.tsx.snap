// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Tag /> > should match snapshot 1`] = `
<div
  class="sc-dcJsrY QRaBo"
  role="tag-container"
>
  destaque
</div>
`;

exports[`<Tag /> should match snapshot 1`] = `
<div
  class="sc-dcJsrY QRaBo"
  role="tag-container"
>
  destaque
</div>
`;
